### PR TITLE
bump compatible Coq version to 8.18~

### PR DESCRIPTION
**Pull request by haselwarter**

### CODE REVIEW

File: .github/workflows/docker-action.yml [hunk 1]
```diff
@@ -18,17 +18,22 @@ jobs:
...
           - 'mathcomp/mathcomp:1.14.0-coq-8.15'
           - 'mathcomp/mathcomp:1.13.0-coq-8.15'
           - 'mathcomp/mathcomp:1.12.0-coq-8.14'
       fail-fast: false
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - uses: coq-community/docker-coq-action@v1
         with:
-          opam_file: 'coq-autosubst-ci.opam'
```
Comment (haselwarter) [inline .github/workflows/docker-action.yml:29]: This change looks a bit suspicious. It is the result of running `generate.sh` from coq-community/templates.

File: .github/workflows/docker-action.yml [hunk 2]
```diff
@@ -18,17 +18,22 @@ jobs:
...
+          opam_file: 'coq-autosubst.opam'
```
Comment (palmskog) [inline .github/workflows/docker-action.yml]:
```suggestion
          opam_file: 'coq-autosubst-ci.opam'
```
If you use the standard opam file, `coq-autosubst.opam`, then the test suite will not be run in CI. For example, this means that the SSR examples are not checked.

**Comment by RalfJung** [comment]
Strange, it builds with coq.dev but on 8.17 we get a deprecation *error* (not warning)?

**Comment by palmskog** [comment]
@RalfJung they turned the `deprecated-instance-without-locality` warning into an error in 8.17. All you need to do is add this flag in the right place (which I'm not sure where it is):
```
-arg -w -arg -deprecated-instance-without-locality
```

**Comment by palmskog** [comment]
And the reason it works on Coq `master` is that it has a new definition of default locality.

**Comment by RalfJung** [comment]
We should probably add the missing `Global` everywhere...

**Comment by RalfJung** [comment]
Thanks!

**Pull request merged**